packages/styles/src/use-style-injection/InsertionPointProvider.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, ReactNode, useMemo, useContext } from "react";
+
+export interface InsertionPointContextType {
+  insertionPoint: ChildNode | null;
+}
+
+const insertionPointContext = createContext<ChildNode | null>(
+  window.document.head.firstChild
```

Comment:
is there a potential danger here if `window.document.head.firstChild` changes or is removed?

I think it might be better to have null as the default and in `useStyleInjection` something like:

```
targetWindow.document.head.insertBefore(
      styleMap.styleElement,
      insertionPoint || targetWindow.document.head.firstChild
    );
```